src/treeview_help.ml

```diff
@@ -0,0 +1,93 @@
+let discord_item ~extension_path =
+  let icon =
+    `LightDark
+      Vscode.TreeItem.LightDarkIcon.
+        { light = `String (extension_path ^ "/assets/discord-light.svg")
+        ; dark = `String (extension_path ^ "/assets/discord-dark.svg")
+        }
+  in
+  let label = Vscode.TreeItemLabel.create ~label:"Chat on Discord" () in
+  let item = Vscode.TreeItem.make ~label () in
+  let command =
+    Vscode.Command.create ~title:"Open" ~command:"vscode.open"
+      ~arguments:
+        [ Vscode.Uri.parse "https://discord.gg/cCYQbqN" () |> Vscode.Uri.t_to_js
+        ]
+      ()
+  in
+  Vscode.TreeItem.set_iconPath item icon;
+  Vscode.TreeItem.set_command item command;
+  item
+
+let discuss_item ~extension_path =
+  let icon =
+    `LightDark
+      Vscode.TreeItem.LightDarkIcon.
+        { light = `String (extension_path ^ "/assets/chat-light.svg")
+        ; dark = `String (extension_path ^ "/assets/chat-dark.svg")
+        }
+  in
+  let label =
+    Vscode.TreeItemLabel.create ~label:"Ask a question on Discuss" ()
+  in
+  let item = Vscode.TreeItem.make ~label () in
+  let command =
+    Vscode.Command.create ~title:"Open" ~command:"vscode.open"
+      ~arguments:
+        [ Vscode.Uri.parse "https://discuss.ocaml.org/" () |> Vscode.Uri.t_to_js
+        ]
+      ()
+  in
+  Vscode.TreeItem.set_iconPath item icon;
+  Vscode.TreeItem.set_command item command;
+  item
+
+let github_item ~extension_path =
+  let icon =
+    `LightDark
+      Vscode.TreeItem.LightDarkIcon.
+        { light = `String (extension_path ^ "/assets/github-light.svg")
+        ; dark = `String (extension_path ^ "/assets/github-dark.svg")
+        }
+  in
+  let label = Vscode.TreeItemLabel.create ~label:"Open an issue on Github" () in
+  let item = Vscode.TreeItem.make ~label () in
+  let command =
+    Vscode.Command.create ~title:"Open" ~command:"vscode.open"
+      ~arguments:
+        [ Vscode.Uri.parse "https://github.com/ocamllabs/vscode-ocaml-platform"
+            ()
+          |> Vscode.Uri.t_to_js
+        ]
+      ()
+  in
+  Vscode.TreeItem.set_iconPath item icon;
+  Vscode.TreeItem.set_command item command;
+  item
+
+let items ~extension_path =
+  [ discord_item ~extension_path
+  ; discuss_item ~extension_path
+  ; github_item ~extension_path
+  ]
+
+let getTreeItem ~extension_path:_ ~element = Promise.return element
+
+let getChildren ~extension_path ~element =
+  match element with
+  | None -> `Value (Some (items ~extension_path))
+  | Some _ -> `Value (Some [])
+
+let register extension =
+  let extension_path = Vscode.ExtensionContext.extensionPath extension in
+  let treeDataProvider =
+    Vscode.TreeDataProvider.create
+      ~getTreeItem:(getTreeItem ~extension_path)
+      ~getChildren:(getChildren ~extension_path)
+      ()
+  in
+  let disposable =
+    Vscode.Window.registerTreeDataProvider ~viewId:"ocaml-help"
+      ~treeDataProvider
+  in
+  Promise.return disposable
```

Comment:
IIUC, even though this function returns a promise, it runs synchronously unless you use `Promise.make` not `Promise.return`, which just wraps the value in a promise and nothing more. Was this the intention?

Reply:
Same for `register` in other treeview_... files.

Reply:
`Promise.make` has type `(resolve:('a -> unit) -> reject:('e -> unit) -> unit) -> 'a Promise.t`, so it wouldn't work in this case (can't return the disposable), unless I'm missing something?

Reply:
I guess we could pass a callback to subscribe instead. I'll try that.

Reply:
Seems I misunderstood how to use `Promise.make`, my bad 🙂

Reply:
If we're sure things we're doing are thread-safe, then I think you could do :

```ocaml
let register extension = 
  Promise.make @@ fun ~resolve ~reject -> 
  let extension_path = Vscode.ExtensionContext.extensionPath extension in
  let treeDataProvider =
    Vscode.TreeDataProvider.create
      ~getTreeItem:(getTreeItem ~extension_path)
      ~getChildren:(getChildren ~extension_path)
      ()
  in
  Vscode.Window.registerTreeDataProvider ~viewId:"ocaml-help" ~treeDataProvider
  |> resolve
```

Reply:
> Seems I misunderstood how to use Promise.make, my bad 🙂
Not at all! Using promises API can be tricky at the beginning. 
I found [this](https://www.npmjs.com/package/bs-promise) to be helpful to see how to create promises.